### PR TITLE
Use Vendetta for dependency management

### PR DESCRIPTION
**Pull request by lclc**
Vendetta is a minimal tool for managing the dependencies in the vendor directory of a go project.
It adds all Go packages as git submodules, to make sure updates of Go packages don't break the build.

See: https://github.com/dpw/vendetta

Built on top of https://github.com/joushou/sshmuxd/pull/14, but I can also do it without the dependencies from there.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,20 +7,25 @@ So, why not just a jump host? Well, if it's just you and no one else needing acc
 Thinking it could be done simpler, sshmux and sshmuxd got written. It allows you to have a proxy that will *only* permit forwarding to user-specific servers, regardless of method. No other poking around is possible, and no having to allow actual login for anyone to the server running sshmuxd.
 
 # Installation
-sshmuxd can be installed from source (super simple with Go).
-To download source:
+The safest way to install sshmuxd is via github:
 
-	go get github.com/joushou/sshmuxd
+	cd $GOPATH/src/github.com/ && mkdir -p joushou && cd joushou
+	git clone https://github.com/joushou/sshmuxd.git && cd sshmuxd
```
Comment (kennylevinsen) [inline README.md]: If you do the following, you don't need "mkdir", a "cd", or the submodule update you added later:
```shell
git clone --recursive https://github.com/joushou/sshmuxd.git $GOPATH/src/github.com/joushou/sshmuxd
```

File: README.md [hunk 2]
```diff
@@ -7,20 +7,25 @@ So, why not just a jump host? Well, if it's just you and no one else needing acc
...
-As long as $GOPATH/bin is in your PATH, you can run it with:
+### Install the dependencies:
 
-	sshmuxd conf.json
-
-Otherwise:
+Dependencies are vendored as git submodules. Run git the following command to fetch them:
 
 	cd $GOPATH/src/github.com/joushou/sshmuxd
+	git submodule update --init --recursive
```
Comment (kennylevinsen) [inline README.md]: If you add `--recursive` to the clone, this becomes unnecessary.

File: README.md [hunk 3]
```diff
@@ -7,20 +7,25 @@ So, why not just a jump host? Well, if it's just you and no one else needing acc
...
+
+### Build sshmuxd:
+
 	go build
-	./sshmuxd example_conf.json
+
+### Install sshmuxd:
+
+	go install
```
Comment (kennylevinsen) [inline README.md]: go build is implicit in go install - maybe just let that be the only step?

File: .gitmodules [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	url = https://github.com/spf13/cast
+[submodule "vendor/github.com/spf13/jwalterweatherman"]
+	path = vendor/github.com/spf13/jwalterweatherman
+	url = https://github.com/spf13/jwalterweatherman
+[submodule "vendor/github.com/spf13/pflag"]
+	path = vendor/github.com/spf13/pflag
+	url = https://github.com/spf13/pflag
+[submodule "vendor/gopkg.in/yaml.v2"]
+	path = vendor/gopkg.in/yaml.v2
+	url = https://gopkg.in/yaml.v2
```
Comment (kennylevinsen) [inline .gitmodules:48]: Damn, that's a long list. I guess this is pulled in by viper? Not complaining, just... Damn.

**Review by kennylevinsen — CHANGES_REQUESTED** [review]
Just a minor comment on the README. Will also merge the other PR first, as this contains both commits.

File: README.md (hunk 1 above)
Comment (kennylevinsen) [reply README.md]: Note that I think the "cd to final sshmuxd folder" kinda belongs down in the build/install section.

**Comment by kennylevinsen** [comment]
I'm entirely fine with using vendored submodules. I didn't previously because my dependencies where quite minimal (and I was lazy at the time - work and all). As long as the tool is merely an external convenience that users/developers won't *need* to use, then I'm all okay with it.

**Comment by lclc** [comment]
Thanks for the feedback, I've changed the README accordingly.

I plan to update & rebase this PR again after the other gets merged.

**Comment by kennylevinsen** [comment]
Cool, I'll merge after the other one. Git should be smart enough to not need a rebase.

**Review by kennylevinsen — APPROVED** [review]

**Pull request merged**